lib/src/Base/IterativeStat/IterativeMoments.cxx

```diff
@@ -0,0 +1,301 @@
+{
+  return new IterativeMoments(*this);
+}
+
+/* String converter */
+String IterativeMoments::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << IterativeMoments::GetClassName()
+      << " iteration=" << iteration_
+      << " dimension=" << dimension_
+      << " order max=" << orderMax_
+      << " centered moments=" << centeredMoments_.__repr__();
+  return oss;
+}
+
+/* String converter */
+String IterativeMoments::__str__(const String & offset) const
+{
+  return getCenteredMoments().__str__(offset);
+}
+
+/*  Accessor to the centered moments */
+Sample IterativeMoments::getCenteredMoments() const
+{
+  return centeredMoments_;
+}
+
+/* Accessor to the maximum order declared */
+UnsignedInteger IterativeMoments::getOrder() const
+{
+  return orderMax_;
+}
+
+Point IterativeMoments::getMomentOne()
+{
+  if (iteration_ >= 1)
+  {
+    return centeredMoments_[0] / iteration_;
+  }
+  else
+  {
+    return centeredMoments_[0];
+  }
+}
+
+/* Accessor to the mean */
+Point IterativeMoments::getMean() const
+{
+  if (!(iteration_ > 0)) throw InternalException(HERE) << "Error: cannot compute the mean of an empty sample.";
+
+  return centeredMoments_[0] / iteration_;
+}
+
+/* Accessor to the variance */
+Point IterativeMoments::getVariance() const
+{
+  if (!(iteration_ > 0)) throw InternalException(HERE) << "Error: cannot compute the variance per component of an empty sample.";
+  if (!(orderMax_ >= 2)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum should be at least 2.";
+
+  /* Special case for a size 1 */
+  if (iteration_ == 1) return Point(dimension_, 0.0);
+
+  /* unbiased estimator */
+  return centeredMoments_[1] / (iteration_ - 1);
+}
+
+/* Accessor to the skewness */
+Point IterativeMoments::getSkewness() const
+{
+  if (!(iteration_ >= 2)) throw InternalException(HERE) << "Error: cannot compute the skewness per component of a sample of size less than 2.";
+  if (!(orderMax_ >= 3)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order should be at least 3.";
+
+  /* Special case for a size 2 */
+  if (iteration_ == 2) return Point(dimension_, 0.0);
+
+
+  Point result(dimension_);
+  const Point varianceEstimator(getVariance());
+  /* unbiased estimator */
+  for(UnsignedInteger d = 0; d < dimension_; ++d)
+  {
+    result[d] = iteration_ / ((iteration_ - 1.0) * (iteration_ - 2.0)) * centeredMoments_(2, d) / std::pow(varianceEstimator[d], 1.5) ;
+  }
+  return result;
+}
+
+/* Accessor to the kurtosis */
+Point IterativeMoments::getKurtosis() const
+{
+  if (!(iteration_ >= 3)) throw InternalException(HERE) << "Error: cannot compute the kurtosis per component of a sample of size less than 3.";
+  if (!(orderMax_ >= 4)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order is lower than 4.";
+
+  /* Special case for a size 3 */
+  if (iteration_ == 3) return Point(dimension_, 0.0);
```

Comment:
This statement is suspicious. Consider a Normal distribution and generate 3 observations. The mean of the sample kurtosis is not equal to 3 (because the estimator is biased), but it is unlikely to be zero. I suggest to remove this special case and avoid returning a value if the regular calculation cannot be performed.